lib/arel/visitors/to_sql.rb

```diff
@@ -170,7 +170,9 @@ def visit_Arel_Nodes_False o, collector
       end
 
       def table_exists? name
-        schema_cache.table_exists? name
+        schema_cache.respond_to?(:data_source_exists?) ? 
```

Comment:
We don't need to check if the method exist. arel 7.1 and master only works with Rails 5. Can you remove the check?